pkg/pillar/cmd/zedrouter/networkinstance.go

```diff
@@ -1870,3 +1860,178 @@ func uplinkToPhysdev(ifname string) string {
 	log.Functionf("uplinkToPhysdev found %s", kernIfname)
 	return kernIfname
 }
+
+func isValidIPRange(ipRange types.IpRange) bool {
+	if ipRange.Start == nil || ipRange.End == nil ||
+		ipRange.Start.IsUnspecified() || ipRange.End.IsUnspecified() {
+		return false
+	}
+	return true
+}
+
+func getIPRangeAddrCount(ipRange types.IpRange) int {
+	var start, end uint
+	start = uint(ipRange.Start[0])<<24 + uint(ipRange.Start[1])<<16 +
+		uint(ipRange.Start[2])<<8 + uint(ipRange.Start[3])
+	end = uint(ipRange.End[0])<<24 + uint(ipRange.End[1])<<16 +
+		uint(ipRange.End[2])<<8 + uint(ipRange.End[3])
+	return int(end - start)
+}
+
+func isValidIPSubnet(subnet net.IPNet) bool {
+	if subnet.IP == nil || subnet.Mask == nil ||
+		subnet.IP.IsUnspecified() {
+		return false
+	}
+	if len(subnet.IP) == 0 || len(subnet.Mask) == 0 {
+		return false
+	}
+	setFlag := false
+	for i := 0; i < len(subnet.IP); i++ {
+		if subnet.IP[i] != 0 {
+			setFlag = true
+			break
+		}
+	}
+	if setFlag == false {
+		return false
+	}
+	setFlag = false
+	for i := 0; i < len(subnet.Mask); i++ {
+		if subnet.Mask[i] != 0 {
+			setFlag = true
+			break
+		}
+	}
+	if setFlag == false {
+		return false
+	}
+	netStr := subnet.String()
+	if netStr == "" {
+		return false
+	}
+	if _, _, err := net.ParseCIDR(netStr); err != nil {
+		return false
+	}
+	return true
+}
+
+func getIPSubnetAddrCount(subnet net.IPNet) int {
+	networkStr := subnet.String()
+	maskLenStr := strings.Split(networkStr, "/")[1]
+	maskLen, _ := strconv.Atoi(maskLenStr)
+	addrcount := 1 << (32 - maskLen)
+	// discount the network, gateway and broadcast addresses(3)
+	addrcount -= 3
+	return addrcount
+}
+
+// do a basic validation for the config
+func isValidIPConfig(ctx *zedrouterContext,
+	status *types.NetworkInstanceStatus) bool {
+	if !isValidIPRange(status.DhcpRange) {
+		return isValidIPSubnet(status.Subnet)
+	}
+	return false
+}
+
+func validateIPConfig(ctx *zedrouterContext,
+	status *types.NetworkInstanceStatus) {
+	status.IsValidIPRange = isValidIPRange(status.DhcpRange)
+	status.IsValidIPSubnet = isValidIPSubnet(status.Subnet)
+	status.IsValidIPConfig = status.IsValidIPSubnet || status.IsValidIPRange
+	publishNetworkInstanceStatus(ctx, status)
+}
+
+// initialize the Host IP Address Management Module
+func handleNetworkInstanceIPAddrInit(ctx *zedrouterContext,
+	status *types.NetworkInstanceStatus) {
+
+	validateIPConfig(ctx, status)
+	if !status.IsValidIPConfig {
+		return
+	}
+	addrcount := 253
```

Comment:
This efault is misleading since it will never be used.

Reply:
sure